### ISSUE TITLE
Relative external modules resolve to the wrong place when CWD too short

### ISSUE THREAD

**Issue by mattdrose** [rollup/rollup#2901]
- Rollup Version: 1.14.2
- Operating System (or Browser): macOS, and Linux
- Node Version: v10.15.2

### How Do We Reproduce?

1. Import a module that jumps up a couple directories
2. Build as an external cjs module
3. Mock `process.cwd()` to have less depth than the directories you jumped back

There's a REPL here: https://repl.it/@mattdrose/CiRollupExample

### Expected Behavior

```javascript
import Foo from '../../foo';
```

becomes

```javascript
const Foo = require('../../foo');
```

### Actual Behavior

```javascript
import Foo from '../../foo';
```

becomes

```javascript
const Foo = require('../foo');
```

### Why is this happening?

I was able to trace the issue to this line:

https://github.com/rollup/rollup/blob/b1df51723fd8c4f951081b20b6cc8f42ef372165/src/Chunk.ts#L624

This resolves to:

```javascript
path.relative('.', '../../foo');
```

Since node's `path.relative` is relative to `process.cwd()`, if `process.cwd()` offers less directories than what you're trying to jump back, it will begin to omit `../`'s.

Having a `cwd` close to `/` is common in CI environments.

**Comment by lukastaegert** [comment]
Thanks, fix at #2901

**Comment by mattdrose** [comment]
Thanks so much for a quick fix 🙌
